### PR TITLE
Run migrations and refresh app state if relevant

### PR DESCRIPTION
**Pull request by TomatoToaster**
CC: @luacmartins (since you worked on this recently) @roryabraham @marcaaron 
### HOLD
- [x] Needs [this PR](https://github.com/Expensify/Web-Expensify/pull/33122) to be on production, or else the new API command will error out.

### Details

We will run FixAccount like we do in OldDot and reinitialize data if something changed for the user or we ran migrations. If FixAccount returns no changes (which it will 90% of the time), we will do nothing. This will work without refreshing the whole app state and users won't need to log out and log back in.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/195477

### Tests
Same as QA but done locally (with the other Web-E PR linked above active)

- [x] Verify that no errors appear in the JS console

### QA Steps
You can reach out to `amal@expensify.com` @TomatoToaster to test this on staging if needed.

1. Find an account that was made before 03/28/2022 (your expensifail should work)
2. If you haven't logged into OldDot since the above date on this account that is ideal.
3. Go to staging.new.expensify.com and open up the dev console and make sure you have verbose enabled:
image

4. Filter the console commands by `FixAccount`
image

5. Now log into your expensifail account. Verify that you see these commands fire off (If you don't see the third one, that's fine for this step, that means you likely logged into this account in OldDot some time after 03/28/2022):
image

6. In another tab, log into the same account on OldDot, and then run this snippet in the console:
`NVP.set('expensify_migration_2022_03_28_policyExpenseChatMigration')` Then refresh the page on NewDot and verify that the three commands in the above screenshot are there.

7. Refresh the page one more time and verify you only see the first 2 commands:
image

- [ ] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

**Comment by TomatoToaster** [comment]
Hmm it does seem to be causing an infinite loop, still need to fix that.

**Comment by TomatoToaster** [comment]
The lint failure is related to this: https://github.com/Expensify/App/pull/8882#issuecomment-1118846252. 

This PR can be reviewed before the hold is lifted and despite that lint fail (which isn't relevant to the files changed here).

**Review by luacmartins — COMMENTED** [review]
Left a couple of comments. I'll test it now!

File: src/libs/actions/App.js
Comment (luacmartins) [inline src/libs/actions/App.js:129]:
```suggestion
            Log.info('FixAccount found updates for this user, so data will be reinitialized', true, response);
```

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (luacmartins) [reply src/libs/Navigation/AppNavigator/AuthScreens.js:111]: That's a good point and I think we should do this instead!

**Comment by luacmartins** [comment]
Tests steps are clear and it seemed to work!

Note: I noticed that [this call](https://github.com/Expensify/Web-Expensify/blob/aed25ec4ea75a9dcf9de9a9996e2481b199bdb65/lib/User/FixAccount.php#L78) always returned a change in my dev account. However, I did verify that the migration ran and the NVP was set.

File: src/libs/actions/App.js
Comment (luacmartins) [reply src/libs/actions/App.js:130]: Ah lol can't believe I made that mistake twice 🤦

Comment (luacmartins) [reply src/libs/actions/App.js:130]: Maybe we should remove `new policies` from the docs then? That threw me off haha

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (TomatoToaster) [reply src/libs/Navigation/AppNavigator/AuthScreens.js:111]: So the reason I did it this way was because we wouldn't have to wait for User_FixAccount _most of the time_ anyway.

Since most of the time fixAccount is going to return `{changed: false}` even if it takes a negligible 50ms (or whatever the user's ping is at the time) to send/receive the response for fixAccount, that's extra time we'll ALWAYS have when waiting for this API command before we fetch the rest of the data.

On the flip side, we'd be making those 42 extra read requests _very rarely_ (pretty much only when we introduce a new migration). For all the times a user refreshes a page, or reopens an an app (not sure how it works on mobile), they'll be making so many more of those 42 read requests, that the rare 84 seems negligible.

I was thinking of it sort of as an optimistic action to just take the initial `App.getData()`.

File: src/libs/actions/App.js
Comment (TomatoToaster) [reply src/libs/actions/App.js:130]: Yeah now that I think about it, it's unlikely a migration will just give a user a new policy. I'll change the docs.

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (marcaaron) [reply src/libs/Navigation/AppNavigator/AuthScreens.js:111]: Ok, I can see how both sides of this argument can be right so probably doesn't matter right now.

**Review by roryabraham — APPROVED** [review]
Late to the party, but LGTM 👍

**Comment by TomatoToaster** [comment]
HOLD will likely be released later today when the next deploy goes up. I'll merge this PR right after that. If there's no other comments, I'll merge it soon after that since we already have the approve and I don't think any other comments are likely to pop up.

**Comment by bondydaa** [comment]
unassigning myself since amal said he will QA

**Comment by roryabraham** [comment]
We've got a bad bug on production right now, and since we can't CP to production ([yet](https://github.com/Expensify/Expensify/issues/195693)), our only way to fix it is by doing another full production deploy. So I'm going to take a crack at the internalQA for this bad boi

**Comment by roryabraham** [comment]
Nice, this was a pass. Going to check it off the checklist

**Pull request merged**
